.github/workflows/docker-build.yaml

```diff
@@ -0,0 +1,49 @@
+name: Docker Build and Push
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+      - docker-build
+    tags:
+      - "*.*.*"
+
+jobs:
+  build-server:
+    name: Docker Build and Push Server
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ secrets.DOCKER_REPO }}/server
+          tags: |
+            type=ref,event=branch
+            type=sha,prefix=
+            type=match,pattern=\d.\d.\d
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ./docker/server/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
Also, can we make Docker use the tag `latest` for our main branch? Then we don't need to specify `stackauth/server:main`

Reply:
done